core/src/main/java/org/apache/iceberg/jdbc/JdbcTable.java

```diff
@@ -0,0 +1,187 @@
+  public static final String SQL_TABLE_NAME = "iceberg_tables";
+  public static final String SQL_TABLE_DDL =
+          "CREATE TABLE " + JdbcTable.SQL_TABLE_NAME +
+                  "(catalog_name VARCHAR(1255) NOT NULL," +
+                  "table_namespace VARCHAR(1255) NOT NULL," +
+                  "table_name VARCHAR(1255) NOT NULL," +
+                  "metadata_location VARCHAR(32768)," +
+                  "previous_metadata_location VARCHAR(32768)," +
+                  "PRIMARY KEY (catalog_name, table_namespace, table_name)" +
+                  ")";
+  public static final String SQL_SELECT = "SELECT * FROM " + SQL_TABLE_NAME +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? ";
+  public static final String SQL_SELECT_ALL = "SELECT * FROM " + SQL_TABLE_NAME +
+          " WHERE catalog_name = ? AND table_namespace = ?";
+  public static final String SQL_UPDATE_METADATA_LOCATION = "UPDATE " + SQL_TABLE_NAME +
+          " SET metadata_location = ? , previous_metadata_location = ? " +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? AND metadata_location = ?";
+  public static final String SQL_INSERT = "INSERT INTO " + SQL_TABLE_NAME +
+          " (catalog_name, table_namespace, table_name, metadata_location, previous_metadata_location) " +
+          " VALUES (?,?,?,?,?)";
+  public static final String SQL_UPDATE_TABLE_NAME = "UPDATE " + SQL_TABLE_NAME +
+          " SET table_namespace = ? , table_name = ? " +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? ";
+  public static final String SQL_DELETE = "DELETE FROM " + SQL_TABLE_NAME +
+          " WHERE catalog_name = ? AND table_namespace = ? AND table_name = ? ";
+
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcTable.class);
+  private final String catalogName;
+  private final Connection dbConn;
+
+  public JdbcTable(Connection dbConn, String catalogName) {
+    this.dbConn = dbConn;
+    this.catalogName = catalogName;
+  }
+
+  public void setAutoCommitOff() throws SQLException {
+    this.dbConn.setAutoCommit(false);
+  }
+
+  public void setAutoCommitOn() throws SQLException {
+    this.dbConn.setAutoCommit(true);
+  }
+
+  public void commit() throws SQLException {
+    this.dbConn.commit();
+  }
+
+  public void rollback() throws SQLException {
+    this.dbConn.rollback();
+  }
+
+  public boolean isExists(TableIdentifier tableIdentifier) {
+    try {
+      return !this.get(tableIdentifier).isEmpty();
+    } catch (SQLException | JsonProcessingException throwables) {
+      return false;
+    }
+  }
+
+  public Map<String, String> get(TableIdentifier tableIdentifier) throws SQLException, JsonProcessingException {
+    Map<String, String> table = Maps.newHashMap();
+    PreparedStatement sql = dbConn.prepareStatement(SQL_SELECT);
+    sql.setString(1, catalogName);
+    sql.setString(2, JdbcUtil.namespaceToString(tableIdentifier.namespace()));
```

Comment:
Rather than using JSON to encode the namespace, I suggest converting it to a String using `Joiner.on(".")`.

There's a trade-off to simplifying the problem by doing it that way: both ```a.`b.c` ``` and `a.b.c` end up as `a.b.c`. That has two effects:
1. Namespaces that "collide" like that are considered the same namespace
2. When listing namespace `a` the result is `[b]` and not `[b.c, b]`

I think that those are fine. For the collisions, I think it is rare for users to _want_ namespaces that can collide but are not the same. If someone creates the table ```a.`b.c`.table_name``` then I really doubt that everyone wants to escape `b.c` every time they type the identifier. Everyone is going to type `a.b.c.table_name` anyway, which would return incorrect results if both tables exist. So not allowing namespaces that conflict that that is actually a good thing.

This is exactly what Iceberg does to look up columns in a table, too. All column names are converted to a lookup key by joining the parts with `.`. Columns that have different structures but identical names aren't allowed, and I think it is better that way.

Reply:
changed to Joiner